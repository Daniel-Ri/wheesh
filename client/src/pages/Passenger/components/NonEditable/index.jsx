/* eslint-disable jsx-a11y/label-has-associated-control */
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import './custom.css';

import classes from './style.module.scss';

const NonEditable = ({ passenger }) => (
  <div className={classes.nonEditable}>
    <form>
      <div className={classes.header}>
        <FormattedMessage id="app_personal_information" />
      </div>
      <div className={classes.input}>
        <label htmlFor="gender">
          <FormattedMessage id="app_gender" />
        </label>
        <div className={classes.radios}>
          <div className={classes.radio}>
            <input type="radio" name="gender" value="Male" disabled checked={passenger.gender === 'Male'} />
            <label>
              <FormattedMessage id="app_male" />
            </label>
          </div>
          <div className={classes.radio}>
            <input type="radio" name="gender" value="Female" disabled checked={passenger.gender === 'Female'} />
            <label>
              <FormattedMessage id="app_female" />
            </label>
          </div>
        </div>
      </div>
      <div className={classes.input}>
        <label htmlFor="dateOfBirth">
          <FormattedMessage id="app_date_of_birth" />
        </label>
        <div className={classes.datePickerWrapper}>
          <DatePicker
            name="dateOfBirth"
            selected={new Date(passenger.dateOfBirth)}
            dateFormat="dd/MM/yyyy"
            className={classes.datePicker}
            disabled
          />
        </div>
      </div>

      <div className={classes.header}>
        <FormattedMessage id="app_certificate_information" />
      </div>
      <div className={classes.input}>
        <label htmlFor="idCard">
          <FormattedMessage id="app_id_card" />
        </label>
        <input type="text" name="idCard" id="idCard" value={passenger.idCard} autoComplete="off" disabled />
      </div>
      <div className={classes.input}>
        <label htmlFor="name">
          <FormattedMessage id="app_name" />
        </label>
        <input type="text" name="name" id="name" value={passenger.name} autoComplete="off" disabled />
      </div>

      <div className={classes.header}>
        <FormattedMessage id="app_contact_information" />
      </div>
      <div className={classes.input}>
        <label htmlFor="email">E-mail</label>
        <input type="text" name="email" id="email" value={passenger.email} autoComplete="off" disabled />
      </div>
    </form>
  </div>
);

NonEditable.propTypes = {
  passenger: PropTypes.object.isRequired,
};

export default NonEditable;
