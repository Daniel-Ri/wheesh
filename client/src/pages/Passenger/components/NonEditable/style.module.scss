@import '@styles/base';

.nonEditable {
  width: 100%;
  @include directionY;
  
  form {
    margin-top: 1rem;
    width: 100%;
    @include directionY;

    .header {
      padding: 0.5rem;
      background-color: $color-light-gray;
    }

    label {
      flex: 45%;
    }

    div.input {
      flex: 45%;
      padding: 0.5rem;
      @include directionX;
      justify-content: space-between;
      align-items: center;
    }

    input {
      flex: 45%;
      width: 45%;
      padding: 0.25rem 0.5rem;
      font-size: 80%;
      text-align: right;
      border-radius: 5px;
    }

    input:focus {
      outline: none;
      border-color: $color-picton-blue;
    }

    input:disabled {
      color: #000;
      border-width: 0;
    }

    .radios {
      flex: 45%;
      width: 45%;
      @include directionX;
      justify-content: space-around;
      gap: 1rem;

      .radio {
        @include directionX;
        gap: 0.25rem;
      }
    }

    .datePicker {
      flex: 45%;
      width: 45%;
      align-self: flex-end;
    }

    .verification {
      flex: 45%;
      width: 45%;
      text-align: left;
    }

    .button {
      align-self: flex-end;
      @include directionX;
      justify-content: flex-end;
      flex: 45%;
    }
  }
}