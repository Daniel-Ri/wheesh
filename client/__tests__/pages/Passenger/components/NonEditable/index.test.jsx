import { render } from '@utils/testHelper';
import NonEditable from '@pages/Passenger/components/NonEditable';

let wrapper;

const mockProps = {
  passenger: {
    id: 1,
    userId: 1,
    isUser: true,
    gender: 'Male',
    dateOfBirth: '1995-11-11T17:00:00.000Z',
    idCard: '1234567890123456',
    name: 'John Doe',
    email: 'johndoe@gmail.com',
  },
};

beforeEach(() => {
  wrapper = render(<NonEditable {...mockProps} />);
});

describe('Component NonEditable of Passenger page', () => {
  test('Renders NonEditable component', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('NonEditable')).toBeInTheDocument();
  });

  test('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
