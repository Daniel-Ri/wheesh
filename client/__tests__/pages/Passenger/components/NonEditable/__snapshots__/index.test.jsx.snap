// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component NonEditable of Passenger page Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nonEditable"
        data-testid="NonEditable"
      >
        <form>
          <div
            class="header"
          >
            Informasi Pribadi
          </div>
          <div
            class="input"
          >
            <label
              for="gender"
            >
              Jenis Kelamin
            </label>
            <div
              class="radios"
            >
              <div
                class="radio"
              >
                <input
                  checked=""
                  disabled=""
                  name="gender"
                  type="radio"
                  value="Male"
                />
                <label>
                  Pria
                </label>
              </div>
              <div
                class="radio"
              >
                <input
                  disabled=""
                  name="gender"
                  type="radio"
                  value="Female"
                />
                <label>
                  Wanita
                </label>
              </div>
            </div>
          </div>
          <div
            class="input"
          >
            <label
              for="dateOfBirth"
            >
              Tanggal Kelahiran
            </label>
            <div
              class="datePickerWrapper"
            >
              <div
                class="react-datepicker-wrapper"
              >
                <div
                  class="react-datepicker__input-container"
                >
                  <input
                    class="datePicker"
                    disabled=""
                    name="dateOfBirth"
                    type="text"
                    value="12/11/1995"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="header"
          >
            Informasi Sertifikat
          </div>
          <div
            class="input"
          >
            <label
              for="idCard"
            >
              Kartu Identitas
            </label>
            <input
              autocomplete="off"
              disabled=""
              id="idCard"
              name="idCard"
              type="text"
              value="1234567890123456"
            />
          </div>
          <div
            class="input"
          >
            <label
              for="name"
            >
              Nama
            </label>
            <input
              autocomplete="off"
              disabled=""
              id="name"
              name="name"
              type="text"
              value="John Doe"
            />
          </div>
          <div
            class="header"
          >
            Informasi Kontak
          </div>
          <div
            class="input"
          >
            <label
              for="email"
            >
              E-mail
            </label>
            <input
              autocomplete="off"
              disabled=""
              id="email"
              name="email"
              type="text"
              value="johndoe@gmail.com"
            />
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nonEditable"
      data-testid="NonEditable"
    >
      <form>
        <div
          class="header"
        >
          Informasi Pribadi
        </div>
        <div
          class="input"
        >
          <label
            for="gender"
          >
            Jenis Kelamin
          </label>
          <div
            class="radios"
          >
            <div
              class="radio"
            >
              <input
                checked=""
                disabled=""
                name="gender"
                type="radio"
                value="Male"
              />
              <label>
                Pria
              </label>
            </div>
            <div
              class="radio"
            >
              <input
                disabled=""
                name="gender"
                type="radio"
                value="Female"
              />
              <label>
                Wanita
              </label>
            </div>
          </div>
        </div>
        <div
          class="input"
        >
          <label
            for="dateOfBirth"
          >
            Tanggal Kelahiran
          </label>
          <div
            class="datePickerWrapper"
          >
            <div
              class="react-datepicker-wrapper"
            >
              <div
                class="react-datepicker__input-container"
              >
                <input
                  class="datePicker"
                  disabled=""
                  name="dateOfBirth"
                  type="text"
                  value="12/11/1995"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="header"
        >
          Informasi Sertifikat
        </div>
        <div
          class="input"
        >
          <label
            for="idCard"
          >
            Kartu Identitas
          </label>
          <input
            autocomplete="off"
            disabled=""
            id="idCard"
            name="idCard"
            type="text"
            value="1234567890123456"
          />
        </div>
        <div
          class="input"
        >
          <label
            for="name"
          >
            Nama
          </label>
          <input
            autocomplete="off"
            disabled=""
            id="name"
            name="name"
            type="text"
            value="John Doe"
          />
        </div>
        <div
          class="header"
        >
          Informasi Kontak
        </div>
        <div
          class="input"
        >
          <label
            for="email"
          >
            E-mail
          </label>
          <input
            autocomplete="off"
            disabled=""
            id="email"
            name="email"
            type="text"
            value="johndoe@gmail.com"
          />
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
